pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -28,31 +33,130 @@
+        if isinstance(annotation, at.GroupedMetadata):
+            yield from annotation
+        elif isinstance(annotation, FieldInfo):
+            yield from annotation.metadata
+            # this is a bit problematic in that it results in duplicate metadata
+            # all of our "consumers" can handle it, but it is not ideal
+            # we probably should split up FieldInfo into:
+            # - annotated types metadata
+            # - individual metadata known only to Pydantic
+            annotation = copy(annotation)
+            annotation.metadata = []
+            yield annotation
+        else:
+            yield annotation
+
+
+def apply_known_metadata(annotation: Any, schema: CoreSchema) -> CoreSchema:  # noqa: C901
+    """
+    Apply `annotation` to `schema` if it is an annotation we know about (Gt, Le, etc.).
+    Otherwise return `None`.
+
+    This does not handle all known annotations. If / when it does, it can always
+    return a CoreSchema and return the unmodified schema if the annotation should be ignored.
+
+    Assumes that GroupedMetadata has already been expanded via `expand_grouped_metadata`.
+    """
+    schema = schema.copy()
+    schema_update, _ = collect_known_metadata([annotation])
+    if isinstance(annotation, at.Gt):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.greater_than_validator, gt=annotation.gt),
+                schema,
+            )
+    elif isinstance(annotation, at.Ge):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.greater_than_or_equal_validator, ge=annotation.ge),
+                schema,
+            )
+    elif isinstance(annotation, at.Lt):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.less_than_validator, lt=annotation.lt),
+                schema,
+            )
+    elif isinstance(annotation, at.Le):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.less_than_or_equal_validator, le=annotation.le),
+                schema,
+            )
+    elif isinstance(annotation, at.MultipleOf):
+        if schema['type'] in NUMERIC_SCHEMA_TYPES:
+            schema.update(schema_update)  # type: ignore
+        else:
+            return cs.no_info_after_validator_function(
+                partial(_validators.multiple_of_validator, multiple_of=annotation.multiple_of),
+                schema,
+            )
```

Comment:
could we turns these into a dict and a lookup rather than duplication, no issue if that's not possible or would be uglier.

Reply:
going to leave this as is for now. I think I'll be fighting the type checker to access `annotation.multiple_of` and such if not.